cmd/clusterawsadm/configreader/configreader.go

```diff
@@ -28,21 +28,26 @@ import (
 	bootstrapschemev1 "sigs.k8s.io/cluster-api-provider-aws/cmd/clusterawsadm/api/bootstrap/v1alpha1/scheme"
 )
 
+type errEmptyBootstrapConfig string
+
+func (e errEmptyBootstrapConfig) Error() string {
+	return fmt.Sprintf("bootstrap config file %q was empty", string(e))
+}
+
 // LoadConfigFile loads a YAML file representing a bootstrapv1.AWSIAMConfiguration.
 func LoadConfigFile(name string) (*bootstrapv1.AWSIAMConfiguration, error) {
-	const errFmt = "failed to load bootstrap config file %s, error %v"
 	// compute absolute path based on current working dir
 	iamConfigFile, err := filepath.Abs(name)
 	if err != nil {
-		return nil, fmt.Errorf(errFmt, name, err)
+		return nil, errors.Wrapf(err, "failed to convert IAM config path into absolute path: %s", name)
 	}
 	loader, err := newFsLoader(iamConfigFile)
 	if err != nil {
-		return nil, fmt.Errorf(errFmt, name, err)
+		return nil, errors.Wrap(err, "failed to initialize filesystem loader")
 	}
 	kc, err := loader.Load()
 	if err != nil {
-		return nil, fmt.Errorf(errFmt, name, err)
+		return nil, err
 	}
 	return kc, err
```

Comment:
Any reason not to just change this to:
`return loader.Load()` since we are not wrapping the error or doing anything different in the non-error case.

Reply:
@benmoss , this is the last unresolved conversation, ptal